pkg/vulnsrc/echo/echo.go

```diff
@@ -0,0 +1,143 @@
+package echo
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/samber/oops"
+	bolt "go.etcd.io/bbolt"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/vulnsrc/vulnerability"
+)
+
+type VulnInfo struct {
+	FixedVersion string `json:"fixed_version"`
+	Severity     string `json:"severity"`
+}
+
+type Advisory map[string]VulnInfo
+
+const (
+	echoDir    = "echo"
+	distroName = "echo"
+)
+
+var (
+	source = types.DataSource{
+		ID:   vulnerability.Echo,
+		Name: "Echo",
+		URL:  "https://advisory.echohq.com/data.json",
+	}
+)
+
+type VulnSrc struct {
+	dbc db.Operation
+}
+
+func NewVulnSrc() VulnSrc {
+	return VulnSrc{
+		dbc: db.Config{},
+	}
+}
+
+func (vs VulnSrc) Name() types.SourceID {
+	return source.ID
+}
+
+func (vs VulnSrc) Update(dir string) error {
+	rootDir := filepath.Join(dir, "vuln-list", echoDir)
+	eb := oops.In(string(source.ID)).With("root_dir", rootDir)
+
+	entries, err := os.ReadDir(rootDir)
+	if err != nil {
+		return eb.Wrapf(err, "failed to read directory")
+	}
+
+	advisoryMap := make(map[string]Advisory)
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		filePath := filepath.Join(rootDir, entry.Name())
+		f, err := os.Open(filePath)
+		if err != nil {
+			return eb.With("file_path", filePath).Wrapf(err, "failed to open file")
+		}
+		defer f.Close()
+
+		pkgName := strings.TrimSuffix(entry.Name(), filepath.Ext(entry.Name()))
+		var advisory Advisory
+		if err := json.NewDecoder(f).Decode(&advisory); err != nil {
+			return eb.With("file_path", filePath).Wrapf(err, "json decode error")
+		}
+		advisoryMap[pkgName] = advisory
```

Comment:
You'll still check the recommendations in another cycle later.
Perhaps we can save package here?

Reply:
I'm not sure I understand, you mean to save the package vulnerabilities separately for each package?

Reply:
NWM. I take a look on my idea again and i think your solution is good.